craft_providers/actions/snap_installer.py

```diff
@@ -111,6 +149,21 @@ def inject_from_host(*, executor: Executor, snap_name: str, classic: bool) -> No
     :raises SnapInstallationError: on unexpected error.
     """
     target_snap_path = pathlib.Path(f"/tmp/{snap_name}.snap")
+    host_revision = _get_host_snap_revision(snap_name=snap_name)
+    target_revision = _get_target_snap_revision(snap_name=snap_name, executor=executor)
+    logger.debug(
+        "For snap %r: host revision = %r, target revision = %r",
+        snap_name,
+        host_revision,
+        target_revision,
+    )
+
+    if target_revision is not None and target_revision == host_revision:
+        logger.debug(
+            "Skipping snap injection for snap %r: target is already up-to-date with revision on host.",
```

Comment:
I believe we aren't ending sentences for logs anymore
```suggestion
            "Skipping snap injection for snap %r: target is already up-to-date with revision on host",
```